back/src/Kyoo.RabbitMq/RabbitProducer.cs

```diff
@@ -30,7 +30,8 @@ public RabbitProducer(IConnection rabbitConnection)
 	{
 		_channel = rabbitConnection.CreateModel();
 
-		_channel.ExchangeDeclare("events.resource", ExchangeType.Topic);
+		if (!doesExchangeExist(rabbitConnection, "events.resource"))
+			_channel.ExchangeDeclare("events.resource", ExchangeType.Topic);
```

Comment:
Shouldn't we instead declare and ignore an exception if it is already configured? i think the current implementation can crash due to race conditions.

PS: i'm also fine with merging it as is, even if it crashes due to race migration, the app should restart instantly and use the existing channel.

Reply:
> Shouldn't we instead declare and ignore an exception if it is already configured?

I looked into this approach, but from what I can tell, the RabbitMQ library throws the same exception for most errors returned by the server. This makes it really difficult to tell say, a permissions error and a queue conflict error apart.

You are absolutely right, this is a race condition. Would you prefer that I try to parse the exception message (so that I can tell exception causes apart), or leave this as is? I'd be glad with either.

Reply:
Let's leave it as is then, if it crashes due to a race condition it'll restart anyways.